### DISCUSSION TITLE
User Group - accessing names and handles.

### DISCUSSION THREAD

**Discussion by StuartMcD** [craftcms/cms#14942] [Q&A] [answered]
Currently recording hard coded user group IDs to be saved with entries that I then use to retrieve based on a users group by searching a text field looking for corresponding ID, all is good the world.

`{% set q = 'plainText:3 OR plainText:6' %}`
In reworking this flow, I'm looking to access, programmatically, the user groups detail rather than manually writing case/if statements 

`if 3 { "<h4>Group 3 name</h4>" }
if 6 { "<h4>Group 6 name</h4>" }`

All info uses user.groups to restrict viewing.

Cannot find any documentation about accessing user group names and handles. If I rename a group or introduce more, I'd like to access the names programmatically.

I've found previous discussions but the code used doesn't seem to exist in v4

https://craftcms.stackexchange.com/questions/20575/how-to-list-all-user-groups-and-users

**Comment by brandonkelly** [comment, marked as answer]
You should generally be checking groups based on their handles, rather than IDs. You don’t have any control over user groups’ IDs, and they could change from environment to environment.

> Currently recording hard coded user group IDs to be saved with entries that I then use to retrieve based on a users group by searching a text field looking for corresponding ID, all is good the world.
> 
> `{% set q = 'plainText:3 OR plainText:6' %}`

Is this for a user query? If so you can use the [`group`](https://craftcms.com/docs/5.x/reference/element-types/users.html#group) param to pass in the user group handles you care about:

```twig
{% set users = craft.users()
  .group(['groupA', 'groupB'])
  .all() %}
```

> In reworking this flow, I'm looking to access, programmatically, the user groups detail rather than manually writing case/if statements
> 
> `if 3 { "<h4>Group 3 name</h4>" } if 6 { "<h4>Group 6 name</h4>" }`
>
> All info uses user.groups to restrict viewing.
>
> Cannot find any documentation about accessing user group names and handles. If I rename a group or introduce more, I'd like to access the names programmatically.

[`user.groups`](https://docs.craftcms.com/api/v5/craft-elements-user.html#property-groups) is going to give you an array of the [`UserGroup`](https://docs.craftcms.com/api/v5/craft-models-usergroup.html) models that the user belongs to, and you can access their [`name`](https://docs.craftcms.com/api/v5/craft-models-usergroup.html#property-name) and [`handle`](https://docs.craftcms.com/api/v5/craft-models-usergroup.html#property-handle) from there.

```twig
{% for group in user.groups %}
  {{ group.name|t }}
{% endfor %}
```

You could also call [`user.isInGroup()`](https://docs.craftcms.com/api/v5/craft-elements-user.html#method-isingroup) if you want to restrict certain output unless a user belongs in a particular group.

```twig
{% if user.isInGroup('groupA') %}
  ...
{% endif %}
```
